couch/tests/conftest.py

```diff
@@ -47,24 +48,14 @@ def couch_cluster():
     env['COUCH_PORT'] = common.PORT
     couch_version = env["COUCH_VERSION"][0]
 
-    args = [
-        "docker-compose",
-        "-f", os.path.join(common.HERE, 'compose', 'compose_v{}.yaml'.format(couch_version))
-    ]
-
-    subprocess.check_call(args + ["down"])
-    subprocess.check_call(args + ["up", "-d"])
-
-    # wait for the cluster to be up before yielding
-    if not wait_for_couch():
-        subprocess.check_call(args + ["down"])
-        raise Exception("couchdb container boot timed out!")
-
-    generate_data(couch_version)
-
-    yield
-
-    subprocess.check_call(args + ["down"])
+    with docker_run(
+            compose_file=os.path.join(common.HERE, 'compose', 'compose_v{}.yaml'.format(couch_version)),
+            env_vars=env,
+            conditions=[WaitFor(wait_for_couch)],
```

Comment:
Let's remove the function wait_for_couch then:

```suggestion
            endpoints=[common.URL],
```